src/routes/app/RidesFeed.tsx

```diff
@@ -140,11 +140,31 @@ const RidesFeed = () => {
   }
 
   useEffect(() => {
-    // Logic to sort rides based on sortingOrder
-    let sortedRides = [];
-
-    if (rideData && rideData.getEvents) {
-      sortedRides = [...rideData.getEvents]; // Create a copy to avoid mutating the original state
+    if (rideData && rideData.getEvents && userData && userData.getUser) {
+      let sortedRides = [...rideData.getEvents]; // Create a copy to avoid mutating the original state
+
+      // Gender Restrictions
+      sortedRides = sortedRides.filter(ride => {
+        const userGender = userData.getUser.sex;
+        const privateWomen = ride.privateWomen ?? false; // can this be abused, don't need default if we remove old events
+        const privateNonBinary = ride.privateNonBinary ?? false;
+        const isGenderRestricted = privateWomen || privateNonBinary;
+
+        // men and prefer-not-to-say don't see gender restricted rides
+        if ((userGender === "gender-man" || userGender === "gender-prefer-not-to-say") && isGenderRestricted) {
+          return false;
+        }
+
+        // non-binary users don't see women only rides
+        if (userGender === "gender-non-binary" && (privateWomen && !privateNonBinary)) {
+          return false;
+        }
+
+        // women see all rides
+        return true;
```

Comment:
I think I need to change the AC on the ticket, because we changed the behavior after our meeting with Lincoln and I forgot to update it. Previously, the "private: non-binary" was only visible if they chose "private: women" first. So it made sense for women to see all rides.
However, we changed that so that "private: non-binary" is its own setting, independent of the women one. So in theory there can be rides that only non-binary riders can see (and women can't)